meterfs/meterfs.c

```diff
@@ -49,10 +49,12 @@ static void meterfs_powerctrl(int state, meterfs_ctx_t *ctx)
 
 static int meterfs_fileNameCheck(const char *name)
 {
-	if (*name == '\0'|| strnlen(name, MAX_NAME_LEN + 1) > MAX_NAME_LEN)
+	int len = strnlen(name, MAX_NAME_LEN + 1);
+
+	if (len < 1 || len > MAX_NAME_LEN)
```

Comment:
I think that instead of `MAX_NAME_LEN` it would be better in this case to use `sizeof(((fileheader_t.*)0)->name)`.
```
	const size_t sz = sizeof(((fileheader_t *)0)->name);

	int len = strnlen(name, sz + 1);

	if (len < 1 || len > sz)
		return -EINVAL;
```

Reply:
I avoided using magic define by deleting `meterfs_fileNameCheck()` function and using statements from it directly there, where is access to sizes of adequate header types.